ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/server/execution/ConnectExecutorTest.java

```diff
@@ -161,18 +191,107 @@ public void shouldReturnErrorIfConnectorExists() {
+  }
+
+  @Test
+  public void shouldReturnErrorIfConnectorTypeIsEmpty() {
+    // Given:
+    final CreateConnector createConnectorEmptyType = new CreateConnector(
+        "foo", ImmutableMap.of("connector.class", new StringLiteral(" ")),
+        Type.SOURCE, false);
+    final ConfiguredStatement<CreateConnector> createConnectorEmptyTypeConfigured
+        = ConfiguredStatement.of(PreparedStatement.of(
+        "CREATE SOURCE CONNECTOR foo WITH ('connector.class'=' ');",
+        createConnectorEmptyType), SessionConfig.of(CONFIG, ImmutableMap.of()));
+
+
+    // When:
+    final Optional<KsqlEntity> entity = ConnectExecutor
+        .execute(createConnectorEmptyTypeConfigured, mock(SessionProperties.class),
+            null, serviceContext).getEntity();
+
+    // Then:
+    assertThat("Expected non-empty response", entity.isPresent());
+    assertThat(entity.get(), instanceOf(ErrorEntity.class));
+  }
+
+  private void givenCreationSuccess() {
     when(connectClient.create(anyString(), anyMap()))
         .thenReturn(ConnectResponse.success(
             new ConnectorInfo(
                 "foo",
                 ImmutableMap.of(),
                 ImmutableList.of(),
                 ConnectorType.SOURCE), HttpStatus.SC_OK));
+  }
 
+  private void givenValidationSuccess() {
+    when(connectClient.validate(anyString(), anyMap()))
+        .thenReturn(ConnectResponse.success(
+            new ConfigInfos(
+                "foo",
+                0,
+                ImmutableList.of(),
+                ImmutableList.of()), HttpStatus.SC_OK));
   }
 
-  private void givenError() {
+  private void givenValidationError() {
+    final ConfigInfo configInfo1  = new ConfigInfo(new ConfigKeyInfo("name", "STRING",
+        true, null, "HIGH", "docs",
+            "Common", 1, "MEDIUM", "Connector name",
+        ImmutableList.of()),
+        new ConfigValueInfo("name", null, ImmutableList.of(), ImmutableList.of(
+            "Name is missing"), true));
+    final ConfigInfo configInfo2 = new ConfigInfo(new ConfigKeyInfo("hostname",
+        "STRING", false, null, "HIGH",
+        "docs for hostname",
+        "Common", 2, "MEDIUM", "hostname",
+        ImmutableList.of()),
+        new ConfigValueInfo("hostname", null, ImmutableList.of(),
+            ImmutableList.of("Hostname is required", "Some other error"), true));
+    final ConfigInfo configInfo3 = new ConfigInfo(new ConfigKeyInfo("port",
+        "INT", false, null, "HIGH",
+        "docs for port",
+        "Common", 3, "MEDIUM", "hostname",
+        ImmutableList.of()),
+        new ConfigValueInfo("port", null, ImmutableList.of(), ImmutableList.of(), true));
+    when(connectClient.validate(anyString(), anyMap()))
+        .thenReturn(ConnectResponse.success(
+            new ConfigInfos(
+                "foo",
+                2,
+                ImmutableList.of(),
+                ImmutableList.of(configInfo1, configInfo2, configInfo3)),
+            HttpStatus.SC_BAD_REQUEST));
```

Comment:
Does a validation failure actually result in a 400 return code?

Reply:
Good catch! It's actually 200. Fixed in 77f6de69656c41aac1f463d2c993e797dc1ba4f0.